packages/babel-helper-simple-access/src/index.ts

```diff
@@ -100,12 +100,29 @@ const simpleAssignmentVisitor = {
         return;
       }
 
-      path.node.right = t.binaryExpression(
-        path.node.operator.slice(0, -1),
-        t.cloneNode(path.node.left),
-        path.node.right,
-      );
-      path.node.operator = "=";
+      const operator = path.node.operator.slice(0, -1);
+      if (["&&", "||", "??"].includes(operator)) {
```

Comment:
```suggestion
      if (t.LOGICAL_OPERATORS.includes(operator)) {
```

Reply:
Have pushed a commit making this change.